beetsplug/listenbrainz.py

```diff
@@ -159,35 +156,32 @@ def get_listenbrainz_playlists(self):
             playlist_info = playlist.get("playlist")
             if playlist_info.get("creator") == "listenbrainz":
                 title = playlist_info.get("title")
-                match = re.search(
-                    r"(Missed Recordings of \d{4}|Discoveries of \d{4})", title
+                self._log.debug(f"Playlist title: {title}")
+                playlist_type = (
+                    "Exploration" if "Exploration" in title else "Jams"
                 )
-                if "Exploration" in title:
-                    playlist_type = "Exploration"
-                elif "Jams" in title:
-                    playlist_type = "Jams"
-                elif match:
-                    playlist_type = match.group(1)
-                else:
-                    playlist_type = None
-                if "week of " in title:
+                if "week of" in title:
                     date_str = title.split("week of ")[1].split(" ")[0]
                     date = datetime.datetime.strptime(
                         date_str, "%Y-%m-%d"
                     ).date()
                 else:
-                    date = None
+                    continue
                 identifier = playlist_info.get("identifier")
                 id = identifier.split("/")[-1]
-                if playlist_type in ["Jams", "Exploration"]:
-                    listenbrainz_playlists.append(
-                        {
-                            "type": playlist_type,
-                            "date": date,
-                            "identifier": id,
-                            "title": title,
-                        }
-                    )
+                listenbrainz_playlists.append(
+                    {"type": playlist_type, "date": date, "identifier": id}
+                )
+        listenbrainz_playlists = sorted(
+            listenbrainz_playlists, key=lambda x: x["type"]
+        )
+        listenbrainz_playlists = sorted(
+            listenbrainz_playlists, key=lambda x: x["date"], reverse=True
+        )
```

Comment:
Is the intent here to sort everything by date, and between playlists of the same date, to sort by type?  Setting `key=lambda x: (x["date"], x["type"])` would be more explicit and more efficient.

Reply:
Oh, never mind, I see that sorting one key in reverse might be a bit difficult.